### ISSUE TITLE
Invocation of method with too few positional args does not show error

### ISSUE THREAD

**Issue by spacether** [microsoft/pyright#5261]
Type: <b>Bug</b>

Invocation of method with too few positional args does not show error:

With this codes, the input is an expanded tuple.
Tuples are immutable so pylance knows the tuple length.
The new method is fully defined, so the ide knows the required number of positional arguments.
The passed number of aruments is not enough.
Running the code correctly throws the TypeError:
`TypeError: __new__() missing 1 required positional argument: 'second_required'`

Note: mypy says:
`my_inspect.py:228: error: Missing positional argument "second_required" in call to "frozenset_with_length"`
```
ReqAndOptionalDictPairs = typing.Union[
    typing.Tuple[typing_extensions.Literal['1req'], int],
    typing.Tuple[typing_extensions.Literal['otherReq'], None],
    typing.Tuple[typing_extensions.LiteralString, int]
]

RequiredDictPairs = typing.Union[
    typing.Tuple[typing_extensions.Literal['1req'], int],
    typing.Tuple[typing_extensions.Literal['otherReq'], None]
]

FrozensetInputType = typing.Union[
    frozenset[
        ReqAndOptionalDictPairs
    ],
    frozenset[
        RequiredDictPairs
    ],
]

class frozenset_with_length(frozenset):
    def __new__(
            cls,
            first_required: RequiredDictPairs,
            second_required: RequiredDictPairs,
            *args: ReqAndOptionalDictPairs):
        req_args = [first_required, second_required]
        if not args:
            return super().__new__(cls, *req_args)
        all_args: typing.List[ReqAndOptionalDictPairs] = list(req_args)
        all_args.extend(args)
        return super().__new__(cls, *all_args)

tuple_items: FrozensetInputType = frozenset_with_length(*(
    ('1req', 1),
    # ('otherReq', None),
    # ('a', 1)
))
```

Extension version: 2023.6.10
VS Code version: Code 1.78.2 (Universal) (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:44:45.204Z)
OS version: Darwin x64 21.6.0
Modes:
Sandboxed: Yes

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5250U CPU @ 1.60GHz (4 x 1600)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|10, 11, 9|
|Memory (System)|8.00GB (0.03GB free)|
|Process Argv|--crash-reporter-id c48e9027-7f92-4e80-b41f-684aa4a6e195|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242cf:30382550
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
cmake_vspar411:30581797
vsaa593cf:30376535
pythonvs932:30410667
cppdebug:30492333
vsclangdc:30486549
c4g48928:30535728
dsvsc012:30540252
pynewext54:30695312
azure-dev_surveyone:30548225
2e4cg342:30602488
pyind779:30671433
f6dab269:30613381
pythonsymbol12:30671437
a9j8j154:30646983
showlangstatbar:30737416
vsctsb:30748421
pythonms35:30701012
a2ce3375:30757347
pythonfmttext:30731395
pythoncmvfstrcf:30756944
fixshowwlkth:30730052
showsbindicator:30760979
pythongtdpathcf:30739705
dh2dc718:30763024

```

</details>

**Comment by erictraut** [comment]
Here's a minimal code sample that exhibits the issue:

```python
def func1(a: int, b: int, *args: int): ...
func1(*(1,))
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by spacether** [comment]
Thanks!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.314, which I just published. It will also be included in this week's Pylance prerelease build.
